### ISSUE TITLE
Finding Python errors with backtrace and pdb

### ISSUE THREAD

**Issue by david-morris** [TIBHannover/FAIR-studyGroup#7]
- **What should/will the [learning objective(s)](http://teachtogether.tech/en/process/#s:process-objectives) be?**
  - Install and configure [`backtrace`](https://github.com/nir0s/backtrace)
  - Find one's own code in a `backtrace` callback to debug it
  - Set a trace an access variables in a local context with `pdb.set_trace()`
- **What do participants need to know / install / bring:**
  - Python
  - pip
  - #11  or equivalent

**Comment by rbirkelbach** [comment]
I can also show how to do similar things in R if that's not too much. :)

**Comment by david-morris** [comment]
I think we should see if we have enough material to split it into two lessons to reduce extraneous load and wasted time for people who only use one of the two languages.  

I don't think I learned how to drop into a REPL in R from a function when I was using R, I'd love to hear it!

Also, now that I think of it, I'm wondering if it's possible to react to errors by dropping into a REPL/debug session, like in SBCL (Common Lisp).  That's probably a bit advanced for our audience, though.

**Comment by katrinleinweber** [comment]
Yes, please open a new issue for the same topic in R, regardless of the amount of material. Teaching things both in different languages was criticised by several workshop attendees, so it's probably a good idea to keep study group events specific to one language at a time and maybe schedule such "tandem" events for 2 adjacent weeks.
